src/config.rs

```diff
@@ -5,13 +9,65 @@ pub trait Config {
 
     fn current_shell_name(&self) -> String;
 
-    fn set_current_shell_name(&mut self, name: &str);
+    fn set_current_shell_name(&mut self, name: &str) -> io::Result<()>;
 
-    fn does_shell_exist(&self, name: &str) -> bool;
+    fn does_shell_exist(&self, name: &str) -> io::Result<bool>;
```

Comment:
I thought we were gonna rescind this change?  `Result<bool>` feels kind of weird to me.  The booleans from the std lib seem to be unwrapped for the most part.  Might be interesting to see how that `is_dir` method actually determines if something is a directory and how it handles the potential `Result`s that come from it...